src/lib.rs

```diff
@@ -667,6 +669,34 @@ decl_module! {
 
             Ok(())
         }
+
+        #[weight = FunctionOf(
+            |args: (&T::AccountId, &u8, &Timepoint<T::BlockNumber>)| 500_000 * (*args.1 as u32),
+            DispatchClass::Normal,
+            true
+        )]
+		fn clear(origin, multi_account_id: T::AccountId, max_cancellations: u8, timepoint: Timepoint<T::BlockNumber>) -> DispatchResult {
+			let mut succeeded = 0;
+			let who = ensure_signed(origin)?;
+			let mut cancelled_multisigs = Vec::with_capacity(max_cancellations as usize);
+			let active_multisigs = <Multisigs<T>>::iter_prefix(&who);
+			for m_sig in active_multisigs.take(max_cancellations as usize){
+				ensure!(who == multi_account_id || who == m_sig.depositor, Error::<T>::NotOwner);
+
+				let _ = T::Currency::unreserve(&m_sig.depositor, m_sig.deposit);
+				cancelled_multisigs.push(m_sig);
+				succeeded += 1;
+			}
+
+			<Multisigs<T>>::remove_prefix(&multi_account_id);
+			ensure!(succeeded == max_cancellations, Error::<T>::MultisigCancelIncomplete);
```

Comment:
Nice idea, but this will prevent the cancel events for individual multisigs. Please move this ensure statement behind the loop below.